### ISSUE TITLE
Unable to set the Indian standard time in timezone settings

### ISSUE THREAD

**Issue by pratts** [sequelize/sequelize#11469]
## Issue Description
Hello, we are trying to set the timestamp to IST, i.e. Asia/Kolkata. But, we are getting IANA standard related errors while running queries on TIMESTAMP fields. 

### What are you doing?
'use strict';
const Sequelize = require( 'sequelize');
const config = require("../../bin/config");
const mysql = require('mysql2/promise');

module.exports = {
	connectionPool : new Sequelize(config.cardplayDb.database, config.cardplayDb.user, config.cardplayDb.password, {
		timezone: 'Asia/Kolkata', //We tried, ETC/GMT0+05:30
		host: config.cardplayDb.host,
		dialect: 'mariadb',
		logging: config.cardplayDb.logginng,
		pool: {
			max: 5,
			min: 0,
			idle: 10000
		},
		dialectOptions: {
			useUTC : false
		}
	})
};

```js
// MINIMAL, SELF-CONTAINED code here (SSCCE/MCVE/reprex)
```

### What do you expect to happen?

_I wanted Foo!_

### What is actually happening?

_The output was Bar!_

```
Output here
```

### Environment

- Sequelize version: 5.15.2
- Node.js version: v10.16.3
- Operating System: Centos 7.5
- If TypeScript related: TypeScript version: NA

### How does this problem relate to dialects?

- [ ] I think this problem happens regardless of the dialect.
- [X] I think this problem happens only for the following dialect(s): mariadb
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [X] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by sushantdhiman** [comment]
For `mariadb <= 2.0.3`, there was no support for named timezone. So we had to convert named timezone to their offsets https://github.com/sequelize/sequelize/blob/edf997fa706934612bc07821029bd66b70ab432c/lib/dialects/mariadb/connection-manager.js#L58

As `mariadb >= 2.0.4` now supports named timezones, we can switch this code in question. In your PR, detect mariadb library version. With help of `semver` only run name to offset conversion for `mariadb <= 2.0.3`

**Comment by sushantdhiman** [comment]
Related https://github.com/sequelize/sequelize/issues/10921

**Comment by pratts** [comment]
Hello @sushantdhiman ,
We found another thing.
If we set the timezone through the configurations:
`timezone: 'Asia/Calcutta',
		dialectOptions: {
			timezone: 'Asia/Calcutta',
		},`

Then it queries the database datetime fields correctly.

Can you please tell how both timezone parameters work differently. Because omitting any of them fails the query.

Regards

**Comment by pratts** [comment]
Hello @papb . Thanks for the update.
One more thing. Should the timezone option be changed to 'Asia/Kolkata' in the future ?

**Comment by papb** [comment]
After a brief search it seems that both `Asia/Calcutta` and `Asia/Kolkata` are supposed to be valid, can you try each and see if both work?

**Comment by pratts** [comment]
hello @papb : I tried Asia/Kolkata and it didn't work. Asia/Calcutta did.

**Comment by papb** [comment]
Hmmm, ok, thanks for letting me know. Just to confirm: it only works when you set `Asia/Calcutta` on both options (`timezone` and `dialectOptions`), right? If you remove either option, or change one of them to `Asia/Kalkota`, then it stops working, is this correct?

**Comment by pratts** [comment]
@papb : Yes it is correct. Removing either of the options doesn't work. I need to add it to both options.

**Comment by bharukarupesh11** [comment]
> 

I've tried using the same but it has worked only in 1 direction for me i.e. at the time of writing to the database. **But, while reading from the database it is still reading the timezone incorrectly.** 

I've even referred to this link https://stackoverflow.com/questions/47367893/sequelize-reads-datetime-in-utc-only but still facing an issue while reading from the database in IST('Asia/Calcutta') format.

**Any suggestion, please?**

**Comment by LeonxLJX** [comment]
I'd like to take this one.

Reading the issue, this looks like a documentation-clarity gap rather than a code defect: the current page doesn't give readers enough to self-diagnose, which is why it surfaces here as an issue instead of being answered by the docs.

Plan: locate the relevant section, confirm what the code actually does today (so the docs match behavior, not intent), tighten the wording, and scan neighboring pages for the same inconsistency — otherwise the fix just relocates the confusion.

One thing to confirm before writing: minimal correction scoped to this page, or also cross-link from related guides if the same gap shows up there?

(claiming via @LeonxLJX)
